src/frequenz/sdk/actor/_background_service.py

```diff
@@ -0,0 +1,307 @@
+                while True:
+                    await asyncio.sleep(self._resolution_s)
+                    print(datetime.datetime.now())
+
+        async def main() -> None:
+            # As an async context manager
+            async with Clock(resolution_s=1):
+                await asyncio.sleep(5)
+
+            # Manual start/stop (only use if necessary, as cleanup is more complicated)
+            clock = Clock(resolution_s=1)
+            await clock.start()
+            await asyncio.sleep(5)
+            await clock.stop()
+
+        asyncio.run(main())
+        ```
+
+    Example: Complete example
+        This is the same example as the *Simple example*, but allowing users to specify
+        a custom loop (and name). This option should be preferred when writing
+        background services, to give users full flexibility.
+
+        ```python
+        import datetime
+        import asyncio
+
+
+        class Clock(BackgroundService):
+            def __init__(
+                self,
+                resolution_s: float,
+                *,
+                name: str | None = None,
+                loop: asyncio.AbstractEventLoop | None = None,
+            ) -> None:
+                super().__init__(name=name, loop=loop)
+                self._resolution_s = resolution_s
+
+            async def start(self) -> None:
+                # Note that you should always use the self._loop attribute access loop
+                # functions instead of asyncio directly.
+                self._tasks.add(self.loop.create_task(self._tick()))
+
+            async def _tick(self) -> None:
+                while True:
+                    await asyncio.sleep(self._resolution_s)
+                    print(datetime.datetime.now())
+
+
+        async def main(loop: asyncio.AbstractEventLoop) -> None:
+            print("Clock as async context manager")
+            async with Clock(resolution_s=1, name="Clock with context", loop=loop):
+                await asyncio.sleep(5)
+
+            print("Clock with manual start/stop")
+            clock = Clock(resolution_s=1, name="Manual clock", loop=loop)
+            await clock.start()
+            await asyncio.sleep(5)
+            await clock.stop()
+
+
+        main_loop = asyncio.get_event_loop_policy().new_event_loop()
+        main_loop.run_until_complete(main(main_loop))
+        ```
+    """
+
+    def __init__(
+        self, *, name: str | None = None, loop: asyncio.AbstractEventLoop | None = None
+    ) -> None:
+        """Initialize this BackgroundService.
+
+        Args:
+            name: The name of this background service. If `None`, `str(id(self))` will
+                be used. This is used mostly for debugging purposes.
+            loop: The event loop this background service will run on. If `None`, the
+                current running event loop will be used.
+        """
+        self._name: str = str(id(self)) if name is None else name
```

Comment:
This is what `str` (and `repr`) do. I think having the class name always in `str` and `repr` is a good thing, so it will be duplicate if I add it to the name too.